pkg/catalog/secondary_index_utils.go

```diff
@@ -133,6 +125,12 @@ func IndexParamsToStringList(indexParams string) (string, error) {
 		res += fmt.Sprintf(" %s '%s' ", IndexAlgoParamOpType, opType)
 	}
 
+	if val, ok := result[Async]; ok {
+		if val == "true" {
```

Comment:
Overall probably we should not just compare val with "true".  But here probably OK.